src/auditwheel/repair.py

```diff
@@ -225,13 +225,13 @@ def is_valid_rpath(rpath: str) -> bool:
         )
         for rp in rpaths
     ]
+    if new_rpath is not None:
+        rpaths.append(new_rpath)
     # Remove duplicates while preserving ordering
     # Fake an OrderedSet using OrderedDict
-    rpath_set = OrderedDict([(old_rpath, "") for old_rpath in rpaths])
-    if rpath is not None:
-        rpath_set[rpath] = ""
+    rpaths = list(OrderedDict.fromkeys(rpaths))
```

Comment:
can we just use dict instead here?

Reply:
Since Python 3.6, the `dict` keys are kept in insertion order. But I think explicitly using `OrderedDict` improves readability.

Reply:
ok for now